LocalyticsPlugin/Assets/Localytics/Localytics.cs

```diff
@@ -8,1073 +8,2163 @@
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+			set
+			{
+				#if UNITY_ANDROID
+				LocalyticsClass.CallStatic("setInAppMessageDismissButtonLocation", GetInAppMessageDismissButtonLocationEnum(value));
+				#elif UNITY_IOS
+				_setInAppMessageDismissButtonLocation((uint)value);
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+		public static bool TestModeEnabled
+		{
+			get
+			{
+				#if UNITY_ANDROID
+				return LocalyticsClass.CallStatic<bool>("isTestModeEnabled");
+				#elif UNITY_IOS
+				return _testModeEnabled();
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+			set
+			{
+				#if UNITY_ANDROID
+				LocalyticsClass.CallStatic("setTestModeEnabled", value);
+				#elif UNITY_IOS
+				_setTestModeEnabled(value);
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+#if UNITY_ANDROID
+		
+		public static void ClearInAppMessageDisplayActivity()
+		{
+			LocalyticsClass.CallStatic("clearInAppMessageDisplayActivity");
+		}
+
+		public static void RegisterPush(string senderId)
+		{
+			LocalyticsClass.CallStatic("registerPush", senderId);
+		}
+
+		public static void DisplayPushNotification(Dictionary<string, string> data)
+		{
+			LocalyticsClass.CallStatic("displayPushNotification", DictionaryToBundle(data));
+		}
+
+		public static string PushRegistrationId
+		{
+			get
+			{
+				return LocalyticsClass.CallStatic<string>("getPushRegistrationId");
+			}
+			set
+			{
+				LocalyticsClass.CallStatic("setPushRegistrationId", value);
+			}
+		}
+
+		public static bool NotificationsDisabled
+		{
+			get
+			{
+				return LocalyticsClass.CallStatic<bool>("areNotificationsDisabled");
+			}
+			set
+			{
+				LocalyticsClass.CallStatic("setNotificationsDisabled", value);
+			}
+		}
+#endif
+
+#if UNITY_IOS
+		public static string PushToken
+		{
+			get
+			{
+				return _pushToken();
+			}
+		}
+
+		public static bool InAppAdIdParameterEnabled
+		{
+			get
+			{
+				return _isInAppAdIdParameterEnabled();
+			}
+			set
+			{
+				_setInAppAdIdParameterEnabled(value);
+			}
+		}
+
+#endif
+
+		public static void RegisterForMessagingEvents ()
+		{
+			#if UNITY_ANDROID
+			if (_messagingListener == null)
+			{
+				_messagingListener = new MessagingListener();
+			}
+
+			LocalyticsClass.CallStatic("setMessagingListener", _messagingListener);
+			#elif UNITY_IOS
+			_registerReceiveMessagingCallback (ReceiveMessagingMessage);
+			#else
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+			#endif
+		}
+
+		public static void UnregisterForMessagingEvents ()
+		{
+			#if UNITY_ANDROID
+			if (_messagingListener == null)
+			{
+				return;
+			}
+
+			LocalyticsClass.CallStatic("setMessagingListener", null);
```

Comment:
Not really necessary because the `_messagingListener` is just a bridging class that won't get called at all if we `null` it here, but I'll go ahead and do so.